diff --git a/dist/config/routesConvention.js b/dist/config/routesConvention.js
index 161a769fa22a8565dad5a29aac659f62412e9383..3a4645b591964cc51968a13fb18ed43bf329b73e 100644
--- a/dist/config/routesConvention.js
+++ b/dist/config/routesConvention.js
@@ -84,7 +84,8 @@ function defineConventionalRoutes(appDir, ignoredFilePatterns) {
       let isIndexRoute = routeId.endsWith("/index");
       let fullPath = createRoutePath(routeId.slice("routes".length + 1));
       let uniqueRouteId = (fullPath || "") + (isIndexRoute ? "?index" : "");
-      if (uniqueRouteId) {
+			let isPathlessLayoutRoute = routeId.split("/").pop()?.startsWith("__") === true;
+			if (uniqueRouteId && !isPathlessLayoutRoute) {
         if (uniqueRoutes.has(uniqueRouteId)) {
           throw new Error(`Path ${JSON.stringify(fullPath)} defined by route ${JSON.stringify(routeId)} conflicts with route ${JSON.stringify(uniqueRoutes.get(uniqueRouteId))}`);
         } else {